### DISCUSSION TITLE
Can't extract images for this PDF

### DISCUSSION THREAD

**Discussion by bbfrog** [pymupdf/PyMuPDF#3948] [Q&A] [answered]
### Description of the bug

[Monaleesa_full.pdf](https://github.com/user-attachments/files/17332581/Monaleesa_full.pdf)
Pymupdf can't extract images in page 2 and page 4 of this pdf.

### How to reproduce the bug

import pymupdf
doc = pymupdf.open('Monaleesa_full.pdf')

page_num = 0
for page in doc:
    &nbsp;&nbsp;page_num += 1
    &nbsp;&nbsp;images = page.get_images(full=True)  
    &nbsp;&nbsp;print(f'page {page_num}: {len(images)} images')

### PyMuPDF version

1.24.11

### Operating system

MacOS

### Python version

3.12

**Comment by JorjMcKie** [comment]
Except for page 7 (0-based), none of the pages contains an image.
What you see are vector graphics - no images.

**Comment by JorjMcKie** [comment]
Vector graphics cannot be extracted. All you can do is making a "photo" of the respective page area ...

**Comment by bbfrog** [comment]
Acrobat API can extract the vector graphics and save as png or svg. How does it do this? Is it hard to support in Pymupdf? THanks!

**Comment by JorjMcKie** [comment, marked as answer]
You can try [this](https://github.com/pymupdf/PyMuPDF-Utilities/blob/master/examples/extract-vector-graphics/detect_graphics.py) script. Or do this:
```python
import pymupdf

doc = pymupdf.open("input.pdf")
for page in doc:
    for i, bbox in enumerate(page.cluster_drawings()):
        pix = page.get_pixmap(clip=bbox, dpi=150)
        pix.save(f"{doc.name}-{page.number}-{i}.png")
```

**Comment by bbfrog** [comment]
Thanks @JorjMcKie very much. It works and can extract the image I want. But it also extracted tables from this pdf as drawing, is there any field can differentiate the tables with other drawing? Thanks!

**Reply by JorjMcKie** [reply]
No, the recognition of vector graphics is based on syntax algorithms only - not on a sematic understanding of what it is that is wrapped bey lines, etc.

If however you are lucky enough such that PyMuPDF can identify the tables as tables ... **_then_** you could check whether those table boundary boxes coincide with identified vector graphic boxes. And thus exclude them from picture conversion.

**Reply by bbfrog** [reply]
Thanks. Chatgpt works for identifying whether a image is table or plot:)
